### ISSUE TITLE
NPC Bots bugs out when entering an instance

### ISSUE THREAD

**Issue by Zaila1337** [ZhengPeiRu21/mod-individual-progression#237]
Whenever i'm trying to enter an instance with NPC Bots, they bug out by loosing all their mana and not being able to regen mana anymore until you dismiss them and they get a skull on their target portrait where you should be able to see their level.

![bild](https://github.com/ZhengPeiRu21/mod-individual-progression/assets/38014581/95c827b9-20bf-41ac-bd62-8769a7d28ad0)

The NPC in this case is a hunter and now it's permanently bugged. The other picture is what they look like after i dismissed her completly: 

![bild](https://github.com/ZhengPeiRu21/mod-individual-progression/assets/38014581/d5bbb0bc-61cb-480c-be9a-5435fbe48374)

**Comment by Zaila1337** [comment]
After spending a couple of hours trying to find what could cause this issue, i suspect something is happening between NPCBots and Autobalancer when zoning into instances.

This is a picture of a bots stats directly after hiring it:

![NPCBot Stat before](https://github.com/ZhengPeiRu21/mod-individual-progression/assets/38014581/66e3e2d8-1be8-44cf-800c-622c2078741a)

This is the stats of the exact same bot after zoning into Molten core:

![NPCBot Stat After](https://github.com/ZhengPeiRu21/mod-individual-progression/assets/38014581/46904e1a-5784-4b84-9f1a-735bd404f09e)

The variables **_mana regen_5 no cast_** and **_mana regen_5 casting_** completly dissapeares when zoning into an instance.

**Comment by Kabakaylen** [comment]
NPC bots and Autobalance do not work well together on their own. 

I run this fork https://github.com/trickerer/Autobalance_for_NPCBots and the core with NPC Bots built it https://github.com/trickerer/AzerothCore-wotlk-with-NPCBots

**Comment by dlsmain** [comment]
I'm getting the same issue. It doesn't seem to effect characters that don't use mana though...

**Comment by Badgermilk0** [comment]
I get the issue that Autobalance does not recognize the Bot as a player - so the difficulty curve is not adjusted when running with bots. =(
@ZhengPeiRu21 would it be possible to include the patch Kabakaylen listen in the preview build?

**Comment by SFJake** [comment]
Is there no manual fix for this? Disabling everything in the auto balance config files (both AutoBalance.conf and AutoBalance.conf.dist to be sure).


The enemies are not balanced anymore but the main problem remains: All bots have 0 mana. Bots without mana are unaffected.

**Comment by Sukoshock** [comment]
Struggling with this problem now as well on the update from yesterday, also ran into it on the March version which is why I decided to update. Any bot that uses mana just completely bricks upon entering an instance, and even the bots that don't use mana have their levels bug out. I've found that updating the bots role will fix the level bug, but does not fix the issues regarding mana.

**Comment by ZhengPeiRu21** [comment]
Recent commits should have fixed many issues with NPCbots, Autobalance, and this module. Is anyone continuing to have the problems described in this issue?

**Comment by TwHProx** [comment]
> Recent commits should have fixed many issues with NPCbots, Autobalance, and this module. Is anyone continuing to have the problems described in this issue?

I opened a new issue. The bots lose the mana in the latest update

**Comment by StaticSyntax** [comment]
> Recent commits should have fixed many issues with NPCbots, Autobalance, and this module. Is anyone continuing to have the problems described in this issue?

Yes, I'm still experiencing this issue on the latest release (2024-08-21)

**Comment by Badgermilk0** [comment]
@StaticSyntax 
What Autobalance branch do you use together with NPCbots?

https://github.com/azerothcore/mod-autobalance
or
https://github.com/trickerer/mod-autobalance

**Comment by StaticSyntax** [comment]
> @StaticSyntax What Autobalance branch do you use together with NPCbots?
> 
> https://github.com/azerothcore/mod-autobalance or https://github.com/trickerer/mod-autobalance

I'm not compiling anything myself, I'm just using the stock standard Module Preview Build provided in this repo.

**Comment by Grimfeather** [comment]
outdated

if it's an issue with the latest preview build, it will be fixed in the next one.
preview builds are just to show what this module does, it's not intended to be 100% functional with other modules.
it is encouraged to compile your own server with the modules you want
